src/main/java/project/bookstore/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,100 @@
+package project.bookstore.service.impl;
+
+import jakarta.transaction.Transactional;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Service;
+import project.bookstore.dto.order.CreateOrderRequestDto;
+import project.bookstore.dto.order.OrderDto;
+import project.bookstore.dto.order.PatchOrderDto;
+import project.bookstore.exception.unchecked.EntityNotFoundException;
+import project.bookstore.mapper.OrderMapper;
+import project.bookstore.model.CartItem;
+import project.bookstore.model.Order;
+import project.bookstore.model.OrderItem;
+import project.bookstore.model.ShoppingCart;
+import project.bookstore.model.User;
+import project.bookstore.repository.order.OrderRepository;
+import project.bookstore.repository.shoppingcart.ShoppingCartRepository;
+import project.bookstore.service.OrderService;
+
+@Service
+@RequiredArgsConstructor
+@Transactional
+public class OrderServiceImpl implements OrderService {
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final OrderMapper orderMapper;
+    private final OrderRepository orderRepository;
+
+    @Override
+    public OrderDto createOrder(
+            CreateOrderRequestDto requestDto, Authentication authentication) {
+        User user = (User) authentication.getPrincipal();
+        ShoppingCart cart = shoppingCartRepository.findByUserId(user.getId());
+        if (cart.getCartItems().isEmpty()) {
+            throw new EntityNotFoundException("Cart is empty");
```

Comment:
add parameter to message
```suggestion
            throw new OrderProcessingException("Cart is empty");
```